src/bindings/python/src/openvino/frontend/pytorch/fx_decoder.py

```diff
@@ -217,15 +217,39 @@ def __init__(self, pt_module, fx_gm=None, nodes=None,
             # FIXME: Quadratic complexity nodes*nodes considering the outer loop over all nodes
             self._outputs = [("", self._nodes.index(pt_module))]
 
+            # Check if this is a higher_order op like while_loop
+            target_str = str(pt_module.target) if pt_module.op == "call_function" else ""
+            is_while_loop = "while_loop" in target_str
+
             self.input_types = []
-            for arg in pt_module.args:
-                if isinstance(arg, torch.fx.Node):
-                    self._inputs.append(self._nodes.index(arg))
-                else:
-                    # Not a node, consider it inlined
-                    self._inputs.append(InlinedInput(arg))
-                self.input_types.append(
-                    BaseFXDecoder.get_type_for_value(arg))
+            if is_while_loop:
```

Comment:
Do you really need special handling for inputs in case of loop? Can't you make it universal?